test/framework/setup.go

```diff
@@ -0,0 +1,171 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/retry"
+	kube_core "k8s.io/api/core/v1"
+	kube_meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type InstallFunc func(cluster Cluster) error
+
+func Yaml(yaml string) InstallFunc {
+	return func(cluster Cluster) error {
+		_, err := retry.DoWithRetryE(cluster.GetTesting(), "install yaml resource", defaultRetries, defaultTimeout,
+			func() (s string, err error) {
+				return "", k8s.KubectlApplyFromStringE(cluster.GetTesting(), cluster.GetKubectlOptions(), yaml)
+			})
+		return err
+	}
+}
+
+func YamlPath(path string) InstallFunc {
+	return func(cluster Cluster) error {
+		_, err := retry.DoWithRetryE(cluster.GetTesting(), "install yaml resource by path", defaultRetries, defaultTimeout,
+			func() (s string, err error) {
+				return "", k8s.KubectlApplyE(cluster.GetTesting(), cluster.GetKubectlOptions(), path)
+			})
+		return err
+	}
+}
+
+func Kuma() InstallFunc {
+	return func(cluster Cluster) error {
+		return cluster.DeployKuma()
+	}
+}
+
+func WaitService(namespace, service string) InstallFunc {
+	return func(c Cluster) error {
+		k8s.WaitUntilServiceAvailable(c.GetTesting(), c.GetKubectlOptions(namespace), service, 10, 3*time.Second)
+		return nil
+	}
+}
+
+func WaitNumPods(num int, app string) InstallFunc {
+	return func(c Cluster) error {
+		k8s.WaitUntilNumPodsCreated(c.GetTesting(), c.GetKubectlOptions(),
+			kube_meta.ListOptions{
+				LabelSelector: fmt.Sprintf("app=%s", app),
+			}, num, defaultRetries, defaultTimeout)
+		return nil
+	}
+}
+
+func WaitPodsAvailable(namespace, app string) InstallFunc {
+	return func(c Cluster) error {
+		pods, err := k8s.ListPodsE(c.GetTesting(), c.GetKubectlOptions(namespace),
+			kube_meta.ListOptions{LabelSelector: fmt.Sprintf("app=%s", app)})
+		if err != nil {
+			return err
+		}
+		for _, p := range pods {
+			err := k8s.WaitUntilPodAvailableE(c.GetTesting(), c.GetKubectlOptions(namespace), p.GetName(), defaultRetries, defaultTimeout)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+}
+
+func HttpBin() InstallFunc {
+	return Combine(
+		YamlPath(filepath.Join("testdata", "httpbin.yaml")),
+		WaitService("kuma-test", "httpbin"),
```

Comment:
Unfortunately, `httpbin` is pretty big image (like 200MB). Not sure how appropriate it is to make it part of the CI and for local testing as it will incur a huge time to download it every time.

Reply:
Yes, this is a good point, I don't even use all functionality that provides httpbin. I think I can replace it with something like:
```bash
while true ; do nc -l -p 1500 -c 'echo -e "HTTP/1.1 200 OK\n\n $(date)"'; done
```

Reply:
There is `-k` option available in the `netcat-openbsd` 
https://pkgs.alpinelinux.org/contents?branch=edge&name=netcat-openbsd&arch=x86_64&repo=main

But that is not available out of the box with alpine image. The proposed solution should work IMO.